webClient/src/app/app.component.ts

```diff
@@ -34,93 +37,108 @@ import {Message} from 'primeng/components/common/api';
 export class AppComponent {
   title = 'app';
 
+  get sideLocal(): FTASide {
+    return FTASide.LOCAL;
+  }
+  get sideRemote(): FTASide {
+      return FTASide.REMOTE;
+  }
+
   ftaServiceUrl: string;
 
   protocols: SelectItem[] = [
-
-    {label:'SFTP', value: 'sftp'},
-    {label:'FTP', value:'ftp'}
+    {label: 'SFTP', value: 'sftp'},
+    {label: 'FTP', value: 'ftp'}
   ];
   selectedProtocol: SelectItem = this.protocols[0];
   address: string;
   username: string;
   password: string;
-  
+
   connectionMessages: Message[] = [];
 
+  // Local and remote are from the perspective of the first mainframe you connect to.
+  // From the perspective of the user's browser both connections are remote.
   localConnection: Connection;
+  remoteConnection: Connection;
   connections: Connection[] = [];
+  credentialsSubmitted: boolean;
+
+  response: any;
+
+  constructor(
+    @Inject(DOCUMENT) 
+    private document: any,
+    @Inject(Angular2InjectionTokens.PLUGIN_DEFINITION) 
+    private pluginDefinition: ZLUX.ContainerPluginDefinition,
+    ) {
+
+    const host = this.document.location.hostname;
```

Comment:
The use of document seems to be just to do exactly what window.location.hostname can already do https://developer.mozilla.org/en-US/docs/Web/API/Location